src/bun.js/bindings/bindings.cpp

```diff
@@ -1371,7 +1478,20 @@ std::optional<bool> specialObjectsDequal(JSC__JSGlobalObject* globalObject, Mark
     compareAsNormalValue:
         break;
     }
-
+    // globalThis is only equal to globalThis
+    // NOTE: Zig::GlobalObject is tagged as GlobalProxyType
+    case GlobalObjectType: {
+        if (c1Type != c2Type) return false;
+        auto* g1 = jsDynamicCast<JSC::JSGlobalObject*, JSCell>(c1);
+        auto* g2 = jsDynamicCast<JSC::JSGlobalObject*, JSCell>(c2);
```

Comment:
not sure if it's possible, but let's add

```suggestion
        auto* g2 = jsDynamicCast<JSC::JSGlobalObject*, JSCell>(c2);
        if (!g1 || !g2) {
            return !!g1 == !!g2;
        }
```

Reply:
GlobalProxyType may also be cast into a `JSC::JSGlobalObject`, so we still need to check if `JSTypes` are equal

Reply:
they're different types